### ISSUE TITLE
Latest Serenity BDD conflates theme- and capability-level narratives

### ISSUE THREAD

**Issue by jan-molak** [serenity-bdd/serenity-core#3119]
Serenity BDD version: **3.6.12**
Serenity Cucumber version: 2.6.0
Cucumber version: 6.11.0

The issue was reported initially at https://github.com/serenity-js/serenity-js/issues/1649
Repro: https://github.com/jan-molak/serenity-js-issue-1649/tree/bug/nested-readmes

## Context

Consider a Java Spring Boot project using Serenity BDD and Cucumber, like [this one](https://github.com/jan-molak/serenity-js-issue-1649/tree/bug/nested-readmes).
The `features` directory uses the themes/capabilities/features structure:

<img width="595" alt="Screenshot 2023-04-19 at 22 29 28" src="https://user-images.githubusercontent.com/1089173/233204489-59d38e44-0dd2-4110-b4f5-7bfcb7f8b5d4.png">

Notice that the project contains both **theme-level** and **capability-level** narratives recorded in `readme.md` files, which should be supported as [per the current docs](https://serenity-bdd.github.io/docs/reporting/living_documentation#the-requirements-hierarchy).

Theme-level `readme.md` file has content along those lines:

```md 
Custom title for Theme 3

Some lorem ipsum narrative
```

Capability-level `readme.md` file has content like this:
```md
Custom title for Theme 3 Capability 3

Again some unimportant lorem ipsum-style narrative
```

## What did I expect to happen

I was expecting capability-level narratives to be listed under capabilities rather than themes.

## The problem

It looks like the latest Serenity BDD **3.6.12** conflates theme- and capability-level narratives, as you can see demonstrated in [this demo project](https://github.com/jan-molak/serenity-js-issue-1649/tree/bug/nested-readmes):

### Problem 1 - capabilities with custom narratives are listed under "themes"

"Custom title for Theme 3 Capability 3" is a capability, not a theme. It shouldn't appear under "Themes":

  <img width="1133" alt="Screenshot 2023-04-19 at 22 40 39" src="https://user-images.githubusercontent.com/1089173/233206358-4b6b9c8c-1b03-4aa9-88c8-4e855cdcee71.png">

### Problem 2 - capabilities with custom narratives disappear from "capabilities"

"Custom title for Theme 3 Capability 3" is missing from "Capabilities"

<img width="1132" alt="Screenshot 2023-04-19 at 22 41 45" src="https://user-images.githubusercontent.com/1089173/233206538-593ad238-796d-49fd-a3b7-5e563a10d9c0.png">

### Problem 3 - when all capabilities have custom descriptions, "capabilities" tab disappears

The issue becomes more problematic when all capabilities have custom descriptions.
When that happens, all capabilities are listed under themes, and the capabilities report disappears altogether:

<img width="1234" alt="Screenshot 2023-04-19 at 22 55 41" src="https://user-images.githubusercontent.com/1089173/233208698-91840e2e-90ca-45b5-90ea-cbeef6978657.png">

To simulate this condition:
- leave only `resources/features/Theme3`
- delete other subdirectories under `resources`

<img width="592" alt="Screenshot 2023-04-19 at 22 56 20" src="https://user-images.githubusercontent.com/1089173/233208810-55d7c05f-776d-4867-9673-acf273d8534d.png">

## Reproducing the issue

The code to reproduce the issue is available at https://github.com/jan-molak/serenity-js-issue-1649/tree/bug/nested-readmes

```console
git clone git@github.com:jan-molak/serenity-js-issue-1649.git
cd serenity-js-issue-1649
mvn clean verify
open target/site/serenity/index.html
```

**Comment by wakaleo** [comment]
Fixed in 3.6.23

**Comment by jan-molak** [comment]
All is working fine, thank you!
